packages/python-packages/protocol-stub-generator/protocol/protocol_models.py

```diff
@@ -671,6 +794,398 @@ def to_json(self):
+                            )
+
+                else:
+                    if inner_model:
+                        inner_model.append(Token(" ", TokenKind.Newline))
+                        inner_model.append(
+                            Token(" " * (indent * 4), TokenKind.Whitespace)
+                        )
+                        if isinstance(json_request[i], list) and not isinstance(
+                            json_request[i][0], dict
+                        ):
+                            inner_model.append(Token(i, TokenKind.Comment))
+                        else:
+                            inner_model.append(Token(i + ": {", TokenKind.Comment))
+                        name = i
+                    else:
+                        self.add_new_line()
+                        if isinstance(json_request[i], list) and not isinstance(
+                            json_request[i][0], dict
+                        ):
+                            pass
+                        else:
+                            self.add_whitespace(indent)
+                            self.add_comment(None, i + ": {", None)  # + ": {"
+                        name = i
+                        inner_model = []
+            if isinstance(json_request[i], str):
+                in_model = False
+                if indent == 4:
+                    in_model = True
+                    indent = indent + 1
+                    name = i
+                if inner_model:
+                    inner_model.append(Token(" ", TokenKind.Newline))
+                    inner_model.append(Token(" " * (indent * 4), TokenKind.Whitespace))
+                else:
+                    self.add_new_line()
+                    self.add_whitespace(indent)
+                index = json_request[i].find("(optional)")
+                param = json_request[i].split()
+                if i == "str":
+                    if param[0] == "str":
+                        param[0] = "string"
+                    if param[0] == "any-object":
+                        param[0] = "{" + "}"
+                    m_type, key = get_map_type(yaml, name)
+                    m_type = format_type(m_type)
+                    if not key:
+                        key = self.parameters[0].name
+                    if inner_model:
+                        if index != -1:
+                            inner_model.append(
+                                Token(
+                                    key + "? : Map<string, " + param[0] + ">;",
+                                    TokenKind.Comment,
+                                )
+                            )
+                        else:
+                            inner_model.append(
+                                Token(
+                                    key + ": Map<string, " + param[0] + ">;",
+                                    TokenKind.Comment,
+                                )
+                            )
```

Comment:
we can simplify this to have fewer lines of code. Something like:

```suggestion
                        optional = ""
                        if index != -1:
                            optional = "? "
                        inner_model.append(
                            Token(
                                key + optional + ": Map<string, " + param[0] + ">;",
                                TokenKind.Comment,
                            )
                        )
```

comment applies generally